### ISSUE TITLE
`dynamic-import-chunkname` - Accept real regexes

### ISSUE THREAD

**Issue by fregante** [un-ts/eslint-plugin-import-x#392]
eslint configs are no longer just JSON, so passing a stringified regex is not necessary

https://github.com/un-ts/eslint-plugin-import-x/blob/1089d9ff44a49bbb6a23db54c8b4bef2478ee8c2/docs/rules/dynamic-import-chunkname.md?plain=1#L20

This could be

```js
   webpackChunknameFormat: /[a-zA-Z0-57-9-/_]+/,
```

**Comment by JounQin** [comment]
Do you want to raise a PR for it?

**Comment by JounQin** [comment]
@fregante We're using jsonschema for rule options, so I'm not sure how to support this?

**Comment by fregante** [comment]
I suppose it's not possible unless you change the schema format. Since the flat config, you're no longer limited to just JSON values. I don’t think anything stops plugins from accepting even functions (for example)

**Comment by JounQin** [comment]
Maybe ESLint should change to support for example `zod` or `arktype` schema validation for flat configs instead.

**Comment by JounQin** [comment]
@fregante

From @nzakas:

> Rule options need to be JSON serializable.

This still applies to flat config.

See https://discord.com/channels/688543509199716507/711686958635548723/1387433178515505255
